app/Domain/Integrations/Controllers/IntegrationController.php

```diff
@@ -77,20 +86,36 @@ public function create(): Response
         ]);
     }
 
-    public function store(StoreIntegration $storeIntegration): RedirectResponse
+    public function store(StoreIntegrationRequest $request): RedirectResponse
     {
-        $integration = StoreIntegrationMapper::map($storeIntegration, $this->currentUser);
+        $integration = StoreIntegrationMapper::map($request, $this->currentUser);
         $this->integrationRepository->save($integration);
 
         return Redirect::route(
-            TranslatedRoute::getTranslatedRouteName(
-                request: $storeIntegration,
-                routeName: 'integrations.index'
-            )
+            TranslatedRoute::getTranslatedRouteName($request, 'integrations.index')
+        );
+    }
+    public function storeUrl(StoreIntegrationUrlRequest $request, string $id): RedirectResponse
+    {
+        $integrationUrl = new IntegrationUrl(
```

Comment:
Although it's a "oneline" I think a mapper called `StoreIntegrationUrlMapper` could be nice here.